module/VuFind/src/VuFind/Controller/InstallController.php

```diff
@@ -922,4 +923,57 @@ public function homeAction()
         }
         return $this->createViewModel(['checks' => $checks]);
     }
+
+    /**
+     * Get minimal PHP version required for VuFind to run.
+     *
+     * @return string
+     */
+    protected function getMinimalPhpVersion(): string
+    {
+        $composer = $this->getComposerJson();
+        if (empty($composer)) {
+            throw new \Exception('Cannot find composer.json');
+        }
+        $rawVersion = $composer['require']['php'] ?? '';
+        $version = preg_replace('/[^0-9. ]/', '', $rawVersion);
+        $version = preg_split('/[. ]/', $version);
+        if (count($version) >= 3) {
+            return sprintf('%d.%d.%d', $version[0], $version[1], $version[2]);
+        }
+        $version = $composer['config']['platform']['php'] ?? '';
+        if (!empty($version)) {
+            return $version;
+        }
+        throw new \Exception('Cannot parse PHP version from composer.json');
+    }
+
+    /**
+     * Get minimal PHP version ID required for VuFind to run.
+     *
+     * @return int
+     */
+    protected function getMinimalPhpVersionId(): int
+    {
+        $version = explode('.', $this->getMinimalPhpVersion());
+        return $version[0] * 10000 + $version[1] * 100 + $version[2];
```

Comment:
During normalization, should we pad the array with zeroes to ensure that at least three elements are set? e.g. if the PHP version is just set to "7" we want the array to be `[7, 0, 0]`. (Some cases demonstrating this scenario would be a good addition to your test suite, as you alluded to in your comments).

Reply:
It should never happen in this method. As if there aren't at least 3 parts of version, method `getMinimalPhpVersion` throws an exception. But it could be possible, that in the future we may want to define only one or two parts. So I expanded the ability of `getMinimalPhpVersion` to parse also 1 or 2 parts versions and added more test cases.